lib/build.gradle

```diff
@@ -40,7 +40,7 @@ android {
     externalNativeBuild {
         cmake {
             path file('CMakeLists.txt')
-            version '3.10.2'
+            version '3.22.1'  // EDITME: update with the version of CMake you have installed on your system - must be above cmake_minimum_required specified in CMakeLists.txt
```

Comment:
This should not be necessary, because AndroidStudio is able to automatically download cmake 3.10.2 according to [this](https://developer.android.com/studio/projects/install-ndk) documentation.
Indeed, my system wide installation of cmake is version 3.27.7, but android studio uses the locally installed version 3.10.2.

There really seems to be an issue with the reproducibility of the build process. I will also give compiling it on a clean ubuntu VM a try.

Reply:
On both Windows and Ubuntu I had to download CMake manually, it wouldn't find it otherwise, and NDK wouldn't download it. When I tried to use NDK, I ran into lots of issues, including [this one](https://stackoverflow.com/questions/56228822/ndk-does-not-contain-any-platforms) being the first, so it seems how Android Studio is handling the NDK is currently changing since 2021 and it's not fully stable on all platforms. The solution 
of specifying the CMake version comes from [here](https://stackoverflow.com/questions/60718412/could-not-get-version-from-cmake-dir), but this is indeed to use the system wide installation of CMake. I did not know the NDK was supposed to install everything since it didn't.

What I can do is I can try again with the NDK in Ubuntu now that I have a bit more experience. Any command I should run in particular? I would like to try to compile with CMake 3.10.2 but it's not available on Ubuntu 22.04, or I did not find how. At least from the Ubuntu package manager snap or apt-get install, I did not try to install directly from the CMake github releases. I can try this too, and if it compiles with CMake 3.10.2 (because remember that I updated to the latest libqalculate, which may rely on a newer CMake version), we can safely change the version back to CMake 3.10.2 in my PR.